pkg/providers/v1/aws.go

```diff
@@ -5130,3 +5134,28 @@ func (c *Cloud) describeNetworkInterfaces(nodeName string) (*ec2.NetworkInterfac
 	}
 	return eni.NetworkInterfaces[0], nil
 }
+
+func getRegionFromMetadata(cfg *CloudConfig) (string, error) {
```

Comment:
since this is almost identical to the passage 

https://github.com/kubernetes/cloud-provider-aws/blob/8cdf51280cc494a9fd150d8b3c26706f1235c516/pkg/providers/v1/aws.go#L1300-L1317

could you refactor that to also use this function?